rotkehlchen/chain/ethereum/uniswap/uniswap.py

```diff
@@ -220,6 +244,224 @@ def _get_known_asset_price(
+            else:
+                existing_addresses.append(address)
+                min_end_ts = min(min_end_ts, trades_range[1])
+
+        # Request new addresses' trades
+        if new_addresses:
+            start_ts = Timestamp(0)
+            new_address_trades = self._get_trades_graph(
+                addresses=new_addresses,
+                graph_query=self.graph.query,  # type: ignore
+                start_ts=start_ts,
+                end_ts=to_timestamp,
+            )
+            address_ammtrades.update(new_address_trades)
+
+            # Insert last used query range for new addresses
+            for address in new_addresses:
+                entry_name = f'{UNISWAP_TRADES_PREFIX}_{address}'
+                db_query_update_used_query_range(
+                    name=entry_name,
+                    start_ts=start_ts,
+                    end_ts=to_timestamp,
+                )
+
+        # Request existing DB addresses' trades
+        if existing_addresses and min_end_ts <= to_timestamp:
+            address_new_trades = self._get_trades_graph(
+                addresses=existing_addresses,
+                graph_query=self.graph.query,  # type: ignore
+                start_ts=min_end_ts,
+                end_ts=to_timestamp,
+            )
+            address_ammtrades.update(address_new_trades)
+
+            # Insert last used query range for existing addresses
+            for address in existing_addresses:
+                entry_name = f'{UNISWAP_TRADES_PREFIX}_{address}'
+                db_query_update_used_query_range(
+                    name=entry_name,
+                    start_ts=min_end_ts,
+                    end_ts=to_timestamp,
+                )
+
+        # Insert requested trades ([<AMMTrade>]) in DB as trades ([<Trade>])
+        for address in filter(lambda address: address in address_ammtrades, addresses):
+            address_trades = [
+                trade.get_trade_from_ammtrade()
+                for trade in address_ammtrades[address]
+            ]
+            db_query_add_trades(address_trades)
+
+        # Fetch all DB Uniswap trades within the time range, and sort them desc
+        db_trades: List[Trade] = db_query_get_trades(
+            from_ts=from_timestamp,
+            to_ts=to_timestamp,
+            location=Location.UNISWAP,
+        )
+        db_trades.sort(key=lambda trade: trade.timestamp, reverse=True)
+        for db_trade in db_trades:
+            # Get <AMMTrade> from <Trade>
+            try:
+                ammtrade = get_ammtrade_from_trade(db_trade)
+            except AMMTradeIDError as e:
+                self.msg_aggregator.add_error(
+                    f'Error converting Trade to AMMTrade. Skipping trade. '
+                    f'Error was: {str(e)}',
+                )
+                continue
+            db_address_trades[ammtrade.address].append(ammtrade)
+
+        return dict(db_address_trades)
+
+    @staticmethod
```

Comment:
Why make this into a staticmethod? Why pass the graph_query function as a callable and not just do nothing and use it directly from `self.graph`?

You seem to have a tendency to passing functions around. This is completely unnecessary and makes the code a lot more complicated than it should be. Please don't do that.